pkgs/development/python-modules/mapbox-earcut/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, fetchFromGitHub
+, setuptools
+, numpy
+, pybind11
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "mapbox-earcut";
+  version = "2.2.2";
+
+  src = fetchFromGitHub {
+    owner = "skogler";
+    repo = "mapbox_earcut_python";
+    rev = "257c2c82819acf72b2b757f200c160e55b9427fc";
+    sha256 = "14z2vj13a1qcaqc6jvg00f367la5470bclvfi9mmlvpac3hhhywd";
+    };
+
+    nativeBuildInputs = [ setuptools ];
+    buildInputs = [ pybind11 ];
+    propagatedBuildInputs = [ numpy ];
+
+    # Some test failed, some passed...
+    checkInputs = [ pytestCheckHook ];
+    doCheck = false;
```

Comment:
How many failed? Can we disable them? Why do they fail?

Reply:
Output of pytestCheckPhase:
```
============================= test session starts ==============================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.9.0, pluggy-0.13.1
rootdir: /build/source
collected 13 items

tests/test_earcut.py ......FFFFFF.                                       [100%]

=================================== FAILURES ===================================
___________________________ test_end_index_too_large ___________________________

    def test_end_index_too_large():
        verts = np.array([[0, 0], [1, 0], [1, 1]]).reshape(-1, 2)
        rings = np.array([5])

>       with pytest.raises(ValueError, message="Expecting ValueError"):
E       TypeError: Unexpected keyword arguments passed to pytest.raises: message
E       Use context-manager form instead?

tests/test_earcut.py:77: TypeError
___________________________ test_end_index_too_small ___________________________

    def test_end_index_too_small():
        verts = np.array([[0, 0], [1, 0], [1, 1]]).reshape(-1, 2)
        rings = np.array([2])

>       with pytest.raises(ValueError, message="Expecting ValueError"):
E       TypeError: Unexpected keyword arguments passed to pytest.raises: message
E       Use context-manager form instead?

tests/test_earcut.py:85: TypeError
______________________________ test_end_index_neg ______________________________

    def test_end_index_neg():
        verts = np.array([[0, 0], [1, 0], [1, 1]]).reshape(-1, 2)
        rings = np.array([-1])

>       with pytest.raises(ValueError, message="Expecting ValueError"):
E       TypeError: Unexpected keyword arguments passed to pytest.raises: message
E       Use context-manager form instead?

tests/test_earcut.py:93: TypeError
__________________________ test_rings_not_increasing ___________________________

    def test_rings_not_increasing():
        verts = np.array([[0, 0], [1, 0], [1, 1]]).reshape(-1, 2)
        rings = np.array([3, 0, 3])

>       with pytest.raises(ValueError, message="Expecting ValueError"):
E       TypeError: Unexpected keyword arguments passed to pytest.raises: message
E       Use context-manager form instead?

tests/test_earcut.py:101: TypeError
_______________________________ test_rings_same ________________________________

    def test_rings_same():
        verts = np.array([[0, 0], [1, 0], [1, 1]]).reshape(-1, 2)
        rings = np.array([3, 3])

>       with pytest.raises(ValueError, message="Expecting ValueError"):
E       TypeError: Unexpected keyword arguments passed to pytest.raises: message
E       Use context-manager form instead?

tests/test_earcut.py:109: TypeError
________________________________ test_no_rings _________________________________

    def test_no_rings():
        verts = np.array([[0, 0], [1, 0], [1, 1]]).reshape(-1, 2)
        rings = np.array([])

>       with pytest.raises(ValueError, message="Expecting ValueError"):
E       TypeError: Unexpected keyword arguments passed to pytest.raises: message
E       Use context-manager form instead?

tests/test_earcut.py:125: TypeError
=========================== short test summary info ============================
FAILED tests/test_earcut.py::test_end_index_too_large - TypeError: Unexpected...
FAILED tests/test_earcut.py::test_end_index_too_small - TypeError: Unexpected...
FAILED tests/test_earcut.py::test_end_index_neg - TypeError: Unexpected keywo...
FAILED tests/test_earcut.py::test_rings_not_increasing - TypeError: Unexpecte...
FAILED tests/test_earcut.py::test_rings_same - TypeError: Unexpected keyword ...
FAILED tests/test_earcut.py::test_no_rings - TypeError: Unexpected keyword ar...
========================= 6 failed, 7 passed in 0.20s ==========================
```
... looks like the tests don't fit the current pytest-syntax. Could fix that through patching the tests.

Reply:
i think it is done